hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/DistCpSync.java

```diff
@@ -515,6 +521,26 @@ private DiffInfo getRenameItem(DiffInfo diff, DiffInfo[] renameDiffArray) {
     return null;
   }
 
+  /**
+   * checks if a parent dir is marked deleted as a part of dir rename happening
+   * to a path which is excluded by the the filter.
+   * @return true if it's marked deleted
+   */
+  private boolean isParentOrSelfMarkedDeleted(DiffInfo diff,
+      DiffInfo[] deletedDirDiffArray) {
+    for (DiffInfo item : deletedDirDiffArray) {
+      if (item.getSource().equals(diff.getSource())) {
+        if (diff.getType() == SnapshotDiffReport.DiffType.MODIFY) {
+          return true;
+        }
+      }
+      if (isParentOf(item.getSource(), diff.getSource())) {
```

Comment:
How does this code distinguish between this scenario:

create /foo/dir/f1
delete /foo/dir

And:

create /foo/dir/f1
delete /foo/dir
create /foo/dir  ** This will be a create in the diff be retained by the MODIFY filter above if item.getSource == diff.getSource.
create /foo/dir/f1 ** This second case should be kept, but it will match the parent of the deleted /foo/dir, so what stops it getting excluded?

Reply:
in the later case, "create /foo/dir/f1" won't be reported in the snapshot diff itself as the parent itself is created post the old snapshot.